cpu/mips32r2_common/newlib_syscalls_mips_uhi/syscalls.c

```diff
@@ -0,0 +1,323 @@
+* @return pointer set to address `-1` on failure
+*/
+
+void *_sbrk_r(struct _reent *r, ptrdiff_t incr)
+{
+    unsigned int state = irq_disable();
+    void *res = heap_top;
+    if ((heap_top + incr > &_eheap) || (heap_top + incr < &_sheap)) {
+        r->_errno = ENOMEM;
+        res = (void *)-1;
+    } else {
+        heap_top += incr;
+    }
+    irq_restore(state);
+    return res;
+}
+
+/**
+* @brief Get the process-ID of the current thread
+*
+* @return the process ID of the current thread
+*/
+pid_t _getpid(void)
+{
+    return sched_active_pid;
+}
+
+/**
+* @brief Get the process-ID of the current thread
+*
+* @return the process ID of the current thread
+*/
+pid_t _getpid_r(struct _reent *ptr)
+{
+    (void) ptr;
+    return sched_active_pid;
+}
+
+/**
+* @brief Send a signal to a given thread
+*
+* @param r TODO
+* @param pid TODO
+* @param sig TODO
+*
+* @return TODO
+*/
+__attribute__ ((weak))
+int _kill_r(struct _reent *r, pid_t pid, int sig)
+{
+    (void) pid;
+    (void) sig;
+    r->_errno = ESRCH; /* not implemented yet */
+    return -1;
+}
+
+/**
+* @brief Open a file
+*
+* This is a wrapper around @c _open
+*
+* @param r pointer to reent structure
+* @param name file name to open
+* @param flags flags, see man 3p open
+* @param mode mode, file creation mode if the file is created when opening
+*
+* @return fd number (>= 0) on success
+* @return -1 on error
+*/
+int _open_r(struct _reent *r, const char *name, int flags, int mode)
+{
+    return open(name, flags, mode);
+}
+
+/**
+* @brief Read bytes from an open file
+*
+* This is a wrapper around @c _read
+*
+* @param[in] r pointer to reent structure
+* @param[in] fd open file descriptor obtained from @c open()
+* @param[out] dest destination buffer
+* @param[in] count maximum number of bytes to read
+*
+* @return number of bytes read on success
+* @return -1 on error,
+*/
+_ssize_t _read_r(struct _reent *r, int fd, void *dest, size_t count)
+{
+    return read(fd,dest,count);
+}
+
+/**
+* @brief Write bytes to an open file
+*
+* This is a wrapper around @c _write
+*
+* @param[in] r pointer to reent structure
+* @param[in] fd open file descriptor obtained from @c open()
+* @param[in] src source data buffer
+* @param[in] count maximum number of bytes to write
+*
+* @return number of bytes written on success
+* @return -1 on error
+*/
+_ssize_t _write_r(struct _reent *r, int fd, const void *src, size_t count)
+{
+    int res = write(fd, src, count);
+    return res;
+}
+
+/**
+* @brief Close an open file
+*
+* This is a wrapper around @c _close
+*
+* If this call returns an error, the fd should still be considered invalid and
+* no further attempt to use it shall be made, not even to retry @c close()
+*
+* @param[in] r pointer to reent structure
+* @param[in] fd open file descriptor obtained from @c open()
+*
+* @return 0 on success
+* @return -1 on error
+*/
+int _close_r(struct _reent *r, int fd)
+{
+    int res = close(fd);
```

Comment:
Why not simply return close(x)? (same goes for the other wrapper functions as well)